### PR TITLE
[SYCL] aspect_selector implementation

### PR DESCRIPTION
**Pull request by cperkinsintel**
This implements the SYCL 2020 `aspect_selector`, with the aspect_selector returning a  SYCL2020 Callable device selector. 

Derived from original work at https://github.com/intel/llvm/pull/5696  . 

Tests here: https://github.com/intel/llvm-test-suite/pull/1327

### CODE REVIEW

**Comment by cperkinsintel** [comment]
https://github.com/intel/llvm-test-suite/pull/1327

**Comment by cperkinsintel** [comment]
/verify with https://github.com/intel/llvm-test-suite/pull/1327

File: sycl/test/abi/sycl_symbols_windows.dump [hunk 1]
```diff
@@ -306,18 +291,12 @@
 ??$has_property@Vuse_primary_context@cuda@context@property@_V1@sycl@@@image_plain@detail@_V1@sycl@@IEBA_NXZ
 ??$has_property@Vuse_primary_context@cuda@context@property@_V1@sycl@@@sampler@_V1@sycl@@QEBA_NXZ
 ??$has_property@Vuse_primary_context@cuda@context@property@_V1@sycl@@@stream@_V1@sycl@@QEBA_NXZ
-??0AccessorBaseHost@detail@_V1@sycl@@QEAA@$$QEAV0123@@Z
```
[the remark this answers is not in the document]
Comment (cperkinsintel) [reply sycl/test/abi/sycl_symbols_windows.dump:309]: Good catch. The damn Windows .dll has moved and I ran the ABI update against the old one accidentally. I'll fix.

[the remark this answers is not in the document]
Comment (cperkinsintel) [reply sycl/test/abi/sycl_symbols_windows.dump]: done.

**Comment by pvchupin** [comment]
@againull there are fails in post commit. Can you take a look?

**Comment by againull** [comment]
> @againull there are fails in post commit. Can you take a look?

@cperkinsintel is the author of this PR. @cperkinsintel, could you please take a look at post commit failures after your patch?

**Pull request merged**